server/manager/src/main/java/org/apache/accumulo/manager/Manager.java

```diff
@@ -1014,30 +1014,32 @@ public void run() {
     // ACCUMULO-4424 Put up the Thrift servers before getting the lock as a sign of process health
     // when a hot-standby
     //
+    // Start the Manager's Fate Service
+    fateServiceHandler = new FateServiceHandler(this);
+    managerClientHandler = new ManagerClientServiceHandler(this);
     // Start the Manager's Client service
-    clientHandler = new ManagerClientServiceHandler(this);
     // Ensure that calls before the manager gets the lock fail
-    Iface haProxy = HighlyAvailableServiceWrapper.service(clientHandler, this);
-    Iface rpcProxy = TraceUtil.wrapService(haProxy);
-    final Processor<Iface> processor;
-    if (context.getThriftServerType() == ThriftServerType.SASL) {
-      Iface tcredsProxy = TCredentialsUpdatingWrapper.service(rpcProxy, clientHandler.getClass(),
-          getConfiguration());
-      processor = new Processor<>(tcredsProxy);
-    } else {
-      processor = new Processor<>(rpcProxy);
-    }
+    ManagerClientService.Iface haProxy =
+        HighlyAvailableServiceWrapper.service(managerClientHandler, this);
+
     ServerAddress sa;
     try {
-      sa = TServerUtils.startServer(context, getHostname(), Property.MANAGER_CLIENTPORT, processor,
-          "Manager", "Manager Client Service Handler", null, Property.MANAGER_MINTHREADS,
-          Property.MANAGER_MINTHREADS_TIMEOUT, Property.MANAGER_THREADCHECK,
-          Property.GENERAL_MAX_MESSAGE_SIZE);
-    } catch (UnknownHostException e) {
-      throw new IllegalStateException("Unable to start server on host " + getHostname(), e);
+      TProcessor processor = ThriftProcessorTypes.getManagerTProcessor(fateServiceHandler, haProxy,
+          getContext(), getConfiguration());
+
+      try {
+        sa = TServerUtils.startServer(context, getHostname(), Property.MANAGER_CLIENTPORT,
+            processor, "Manager", "Manager Client Service Handler", null,
+            Property.MANAGER_MINTHREADS, Property.MANAGER_MINTHREADS_TIMEOUT,
+            Property.MANAGER_THREADCHECK, Property.GENERAL_MAX_MESSAGE_SIZE);
+      } catch (UnknownHostException e) {
+        throw new IllegalStateException("Unable to start server on host " + getHostname(), e);
+      }
+      clientService = sa.server;
+      log.info("Started Manager client service at {}", sa.address);
+    } catch (Exception e2) {
+      throw new RuntimeException("Error creating thrift server processor", e2);
```

Comment:
Can this catch something more specific?

Reply:
I was able to remove the Exception declaration from `ThriftProcessorTypes.ProcessorType.getProcessor()` and instead throw an `IllegalArgumentException`. This allowed me to remove this completely in 2bac8d5f30828ada41d60168272b0c6b46802dc5